table/block_based/block_based_table_reader.cc

```diff
@@ -2911,25 +3014,51 @@ Status BlockBasedTable::Get(const ReadOptions& read_options, const Slice& key,
           // the end of the block, i.e. cannot be in the following blocks
           // either. In this case, the seek_key cannot be found, so we break
           // from the top level for-loop.
-          break;
-        }
+          done = true;
+        } else {
+          // Call the *saver function on each entry/block until it returns false
+          for (; biter.Valid(); biter.Next()) {
+            ParsedInternalKey parsed_key;
+            if (!ParseInternalKey(biter.key(), &parsed_key)) {
+              s = Status::Corruption(Slice());
+            }
 
-        // Call the *saver function on each entry/block until it returns false
-        for (; biter.Valid(); biter.Next()) {
-          ParsedInternalKey parsed_key;
-          if (!ParseInternalKey(biter.key(), &parsed_key)) {
-            s = Status::Corruption(Slice());
+            if (!get_context->SaveValue(
+                    parsed_key, biter.value(), &matched,
+                    biter.IsValuePinned() ? &biter : nullptr)) {
+              does_referenced_key_exist = true;
+              done = true;
+              break;
+            }
           }
-
-          if (!get_context->SaveValue(
-                  parsed_key, biter.value(), &matched,
-                  biter.IsValuePinned() ? &biter : nullptr)) {
-            done = true;
-            break;
+          s = biter.status();
+          if (s.ok() && done && block_cache_tracer_) {
+            referenced_data_size = biter.key().size() + biter.value().size();
           }
         }
-        s = biter.status();
+        // Write the block cache access record.
+        if (block_cache_tracer_) {
+          // Avoid making copy of block_key, cf_name and referenced_key when
+          // constructing the access record.
+          BlockCacheTraceRecord access_record(
+              rep_->ioptions.env->NowMicros(),
+              /*block_key=*/"", lookup_data_block_context.block_type,
+              lookup_data_block_context.block_size,
+              rep_->get_cf_id_for_tracing(),
+              /*cf_name=*/"", rep_->get_level_for_tracing(),
+              rep_->get_sst_number_for_tracing(),
+              lookup_data_block_context.caller,
+              lookup_data_block_context.is_cache_hit,
+              lookup_data_block_context.no_insert,
+              /*referenced_key=*/"", referenced_data_size,
```

Comment:
Just to confirm, `referenced_key` will be updated in a future change?

Reply:
It is done in this change. I didn't pass the referenced key here to avoid making a copy of string. The key is passed to WriteBlockAccess as const Slice&.